framework/src/main/java/org/checkerframework/common/wholeprograminference/SceneToStubWriter.java

```diff
@@ -0,0 +1,691 @@
+        } else {
+            return formatAnnotations(aType.tlAnnotationsHere) + basetypeToPrint + " ";
+        }
+    }
+
+    /** Writes an import statement for each annotation used in an {@link AScene}. */
+    private static class ImportDefWriter extends DefCollector {
+
+        /** The writer onto which to write the import statements. */
+        private final PrintWriter printWriter;
+
+        /**
+         * Constructs a new ImportDefWriter, which will run on the given AScene when its {@code
+         * visit} method is called.
+         *
+         * @param scene the scene whose imported annotations should be printed
+         * @param printWriter the writer onto which to write the import statements
+         * @throws DefException if the DefCollector does not succeed
+         */
+        ImportDefWriter(ASceneWrapper scene, PrintWriter printWriter) throws DefException {
+            super(scene.getAScene());
+            this.printWriter = printWriter;
+        }
+
+        /**
+         * Write an import statement for a given AnnotationDef. This is only called once per
+         * annotation used in the scene.
+         *
+         * @param d the annotation definition to print an import for
+         */
+        @Override
+        protected void visitAnnotationDef(AnnotationDef d) {
+            if (!isInternalJDKAnnotation(d.name)) {
+                printWriter.println("import " + d.name + ";");
+            }
+        }
+    }
+
+    /**
+     * Return true if the given annotation is an internal JDK annotations, whose name includes '+'.
+     *
+     * @param annotationName the name of the annotation
+     * @return true iff this is an internal JDK annotation
+     */
+    private static boolean isInternalJDKAnnotation(String annotationName) {
+        return annotationName.contains("+");
+    }
+
+    /**
+     * Print the hierarchy of outer classes up to and including the given class, and return the
+     * number of curly braces to close with. The classes are printed with appropriate opening curly
+     * braces, in standard Java style.
+     *
+     * <p>In an AScene, an inner class name is a binary name like "Outer$Inner". In a stub file,
+     * inner classes must be nested, as in Java source code.
+     *
+     * @param basename the binary name of the class without the package part
+     * @param aClass the AClass for {@code basename}
+     * @param printWriter the writer where the class definition should be printed
+     * @return the number of outer classes within which this class is nested
+     */
+    private static int printClassDefinitions(
+            String basename, AClassWrapper aClass, PrintWriter printWriter) {
+
+        String[] classNames = StringUtils.split(basename, '$');
+
+        for (int i = 0; i < classNames.length; i++) {
+            String nameToPrint = classNames[i];
+            printWriter.print(indents(i));
+            if (aClass.isEnum(nameToPrint)) {
```

Comment:
Related to my other comment, this uses the simple name but doesn't have to.
The likelihood of nested classes of the same name is low, though, so it's OK to document the problem somewhere (probably not here) rather than fixing it.

Reply:
I agree. I documented the risk.